ClosedXML/Excel/Style/XLFillKey.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace ClosedXML.Excel
+{
+    public struct XLFillKey : IEquatable<XLFillKey>
```

Comment:
It would be convenient, indeed, but would encourage a quite inefficient usage. ```XLxxxxKey``` are structs, which means this code

```
var fill = new XLFillKey()
.WithBackgroundColor(a)
.WithPatternType(b)
.WithPatternColor(c);
```

will cause 4 XLFillKey instances created, instead of one. With ```ref struct``` it would be fine, but for now - I'd prefer stick to a 'classic' approach:

```
var fill = new XLFillKey
{
  BackgoundColor = a,
  PatterType = b,
  PatternColor = c
};
```

Reply:
Yes, let's park this until we use ref structs.